storage/mysql/mysql_test.go

```diff
@@ -28,14 +63,253 @@ func TestMain(m *testing.M) {
+				tessera.WithPushback(10),
+			},
+			wantErr: false,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := mysql.New(ctx, testDB, test.opts...)
+			gotErr := err != nil
+			if gotErr != test.wantErr {
+				t.Errorf("got err %v want %v", gotErr, test.wantErr)
+			}
+		})
+	}
+}
+
+func TestReadTile(t *testing.T) {
+	ctx := context.Background()
+	s := newTestMySQLStorage(t, ctx)
+
+	for _, test := range []struct {
+		name                string
+		level, index, width uint64
+		wantErr             bool
+	}{
+		{
+			name:  "0/0/0",
+			level: 0, index: 0, width: 0,
+			wantErr: false,
+		},
+		{
+			name:  "123/456/789",
+			level: 123, index: 456, width: 789,
+			wantErr: false,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := s.ReadTile(ctx, test.level, test.index, test.width)
+			gotErr := err != nil
+			if gotErr != test.wantErr {
+				t.Errorf("got err %v want %v", gotErr, test.wantErr)
+			}
+		})
+	}
+}
+
+func TestReadEntryBundle(t *testing.T) {
+	ctx := context.Background()
+	s := newTestMySQLStorage(t, ctx)
+
+	for _, test := range []struct {
+		name    string
+		index   uint64
+		wantErr bool
+	}{
+		{
+			name:    "0",
+			index:   0,
+			wantErr: false,
+		},
+		{
+			name:    "123456789",
+			index:   123456789,
+			wantErr: false,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := s.ReadEntryBundle(ctx, test.index)
+			gotErr := err != nil
+			if gotErr != test.wantErr {
+				t.Errorf("got err %v want %v", gotErr, test.wantErr)
+			}
+		})
+	}
+}
+
+func TestAdd(t *testing.T) {
+	ctx := context.Background()
+	s := newTestMySQLStorage(t, ctx)
+
+	for _, test := range []struct {
+		name      string
+		entry     []byte
+		wantIndex uint64
+		wantErr   bool
+	}{
+		{
+			name:      "empty string entry",
+			entry:     []byte(""),
+			wantIndex: 0,
+			wantErr:   false,
+		},
+		{
+			name:      "123 string entry",
+			entry:     []byte("123"),
+			wantIndex: 1,
+			wantErr:   false,
+		},
+		{
+			name:      "empty byte",
+			entry:     []byte{},
+			wantIndex: 2,
+			wantErr:   false,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			index, err := s.Add(ctx, tessera.NewEntry(test.entry))
+			gotErr := err != nil
+			if gotErr != test.wantErr {
+				t.Errorf("got err %v want %v", gotErr, test.wantErr)
+			}
+			if index != test.wantIndex {
+				t.Errorf("got index %d want %d", index, test.wantIndex)
+			}
+		})
+	}
+}
+
+func TestParallelAdd(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	s := newTestMySQLStorage(t, ctx)
+
+	for _, test := range []struct {
+		name    string
+		entry   []byte
+		wantErr bool
+	}{
+		{
+			name:    "empty string entry",
+			entry:   []byte(""),
+			wantErr: false,
+		},
+		{
+			name:    "123 string entry",
+			entry:   []byte("123"),
+			wantErr: false,
+		},
+		{
+			name:    "empty byte",
+			entry:   []byte{},
+			wantErr: false,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			for i := 0; i < 655536; i++ {
```

Comment:
Typo - 65536? That's still quite a number of goroutines to run...
What's the goal of this test/which functionality of the MySQL implementation are you wanting to exercise here?